server/plugin.go

```diff
@@ -261,18 +261,18 @@ func (p *Plugin) OnActivate() error {
 		conf.rsaKey = rsaKey
 	})
 
-	bundlePath, err := p.API.GetBundlePath()
+	bundlePath, err := p.client.System.GetBundlePath()
 	if err != nil {
 		return errors.Wrap(err, "couldn't get bundle path")
 	}
 
-	profileImage, err := ioutil.ReadFile(filepath.Join(bundlePath, "assets", "profile.png"))
+	profileImage, err := os.Open(filepath.Join(bundlePath, "assets", "profile.png"))
 	if err != nil {
 		return errors.Wrap(err, "couldn't read profile image")
 	}
 
-	if appErr := p.API.SetProfileImage(botUserID, profileImage); appErr != nil {
-		return errors.Wrap(appErr, "couldn't set profile image")
+	if err = p.client.User.SetProfileImage(botUserID, profileImage); err != nil {
+		return errors.Wrap(err, "couldn't set profile image")
 	}
 
```

Comment:
While we are here, let's make use of https://pkg.go.dev/github.com/mattermost/mattermost-plugin-api@v0.1.2#ProfileImagePath.